src/main/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListener.java

```diff
@@ -97,16 +99,27 @@ public final void onStarted(final Run run, final TaskListener listener) {
     // Process only if job is NOT in blacklist
     if ( DatadogUtilities.isJobTracked(run.getParent().getFullDisplayName()) ) {
       logger.fine("Started build!");
+      Queue.Item item = queue.getItem(run.getQueueId());
 
       // Gather pre-build metadata
       JSONObject builddata = new JSONObject();
       builddata.put("job", DatadogUtilities.normalizeFullDisplayName(jobName)); // string
       builddata.put("number", run.number); // int
       builddata.put("result", null); // null
       builddata.put("duration", null); // null
+      builddata.put("waiting", (System.currentTimeMillis() - item.getInQueueSince()) / DatadogBuildListener.THOUSAND_LONG);
       long starttime = run.getStartTimeInMillis() / DatadogBuildListener.THOUSAND_LONG; // ms to s
       builddata.put("timestamp", starttime); // string
 
+      HashMap<String,String> extraTags = new HashMap<String, String>();
+      try {
+        extraTags = DatadogUtilities.parseTagList(run, listener);
+      } catch (IOException ex) {
+        logger.severe(ex.getMessage());
+      } catch (InterruptedException ex) {
+        logger.severe(ex.getMessage());
+      }
+
```

Comment:
I think we should abstract this to a new method in `DataUtilities.java` so we can use it for future metrics as well. I was playing around with this and added this method:

```
  /**
   * Builds extraTags if any are configured in the Job
   *
   * @param run - Current build
   * @param listener - Current listener
   * @return A {@link HashMap} containing the key,value pairs of tags if any.
   */
  public static HashMap<String,String> buildExtraTags(Run run, TaskListener listener) {
    HashMap<String,String> extraTags = new HashMap<String, String>();
    try {
      extraTags = DatadogUtilities.parseTagList(run, listener);
    } catch (IOException ex) {
      logger.severe(ex.getMessage());
    } catch (InterruptedException ex) {
      logger.severe(ex.getMessage());
    }
    return extraTags;
  }
```

Then we can just add these lines here and on [line 160](https://github.com/DataDog/jenkins-datadog-plugin/pull/81/files/7b70f7995fd732377ae4f7bd6eb4997cfd1f7b4d?diff=split#diff-532720f1b459de301183a6478a8e6fa5R160) as well:
```
      HashMap<String,String> extraTags = new HashMap<String, String>();
      extraTags = DatadogUtilities.buildExtraTags(run, listener);
```